### PR TITLE
Add truncate for long errors

### PR DESCRIPTION
**Pull request by smyzin**
- rechecked errors
- added `parseErrorText` function
- created `ErrorMessage` component that can truncate a message if it doesn't fit into two lines
- added error handler on the `/cast` page

https://user-images.githubusercontent.com/5114069/193244292-8cc6627c-2a24-4a2d-8d57-619deeacd3de.mov

### CODE REVIEW

File: src/components/ui/ErrorMessage.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+export default function ({
+  text,
+  truncated,
+}: {
+  text: string
+  truncated?: boolean
+}) {
+  const [isErrorOpen, setOpenError] = useState(false)
+
+  if (!truncated)
```
Comment (T-Damer) [inline src/components/ui/ErrorMessage.tsx:21]: Why do you check `if (!..)` if you can check `if (...)`? It's easier to read + your PC don't need to convert the thing

File: src/components/ui/ErrorMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    return (
+      <ErrorText visible={!!text} withExclamation>
```
Comment (T-Damer) [inline src/components/ui/ErrorMessage.tsx:23]: Let's extract `!!text` into `const isVisible = !!text`

File: src/components/ui/ErrorMessage.tsx (hunk 1 above)
Comment (smyzin) [reply src/components/ui/ErrorMessage.tsx:21]: Changed it

**Review by T-Damer — APPROVED** [review]

File: src/components/ui/ErrorMessage.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  )
+
+export default function ({
+  text,
+  truncated,
+}: {
+  text: string
+  truncated?: boolean
+}) {
+  const [isErrorOpen, setOpenError] = useState(false)
```
[the remark this answers is not in the document]
Comment (smyzin) [reply src/components/ui/ErrorMessage.tsx:19]: Done

File: src/components/Cast/TextareaInfo.tsx [hunk 4]
```diff
@@ -71,6 +75,11 @@ export default function ({
           </AccentText>
         </div>
       )}
+      {!!error && (
+        <div className={displayTo('md')}>
```
Comment (MixailE) [inline src/components/Cast/TextareaInfo.tsx:79]: Why are you using `displayTo('md')` here?

**Review by MixailE — APPROVED** [review]

File: src/components/Cast/TextareaInfo.tsx (hunk 4 above)
Comment (smyzin) [reply src/components/Cast/TextareaInfo.tsx]: To show small text errors on small screens.

**Review by T-Damer — APPROVED** [review]

**Review by T-Damer — APPROVED** [review]

**Pull request merged**